src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -365,7 +365,28 @@ public AgentSettings LoadSettings()
                 {
                     AgentSettings settings = _store.GetSettings();
                     var credentialManager = HostContext.GetService<ICredentialManager>();
-                    VssCredentials creds = credentialManager.LoadCredentials();
+
+                    // Get the credentials
+                    VssCredentials creds;
+                    while (true)
```

Comment:
We should not have a while loop. Unconfig is best effort. Will this potentially this use the persisted credentials? If so, and then does it have access to remove the agent?